src/coreclr/vm/invokeutil.cpp

```diff
@@ -139,7 +139,9 @@ void InvokeUtil::CopyArg(TypeHandle th, PVOID argRef, ArgDestination *argDest) {
 
     switch (type) {
 #ifdef TARGET_RISCV64
```

Comment:
@shushanhf @LuckyXu-HF AFAIK LoongArch ABI also requires integer scalar arguments to be type-extended. If so, you can re-use this code block, \#ifdef'ing locally for `uint` and `float` which LoongArch handles differently.

Reply:
Yes, the integer types are same.
We also met these cases serveral years ago and fixed these cases for LA64.
These cases are related the CPU-processor's final implementation and calling convention, especially the calling convention might be related with the native compiler's optimized implementation

Is there any test cases which you can test these code?  we have to test and make sure these based on the latest code.

Reply:
You can add a few cases to PrimitiveABI but I presume currently LoongArch codegen is adding type-extending instructions before the value is used like RISC-V, so a value with bad extension coming from InvokeUtil::CopyArg won't fail the test. I don't think there's an automatic way to check it currently, you can add breakpoint and see if the register has the ABI-appropriate extension on callee entry.